### ISSUE TITLE
DistillationColumn's CMO path (use_mesh=False) violates the per-species material balance

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/differentiable-flowsheets#211]
## Summary

`DistillationColumn.__call__(..., use_mesh=False)` returns products whose
per-species flows do not add up to the feed. The *total* flows are right — `D`
and `B` are pinned by the `D_spec`/`B_spec` argument — but the compositions are
far enough off that a component can leave in greater quantity than it entered.

The rigorous MESH path (`use_mesh=True`, the default) is fine.

## Reproduce

```python
import jax; jax.config.update("jax_enable_x64", True)
from difflow import IdealThermo, make_stream, get_flows
from difflow.database import get_species_data
from difflow.units.distillation import DistillationColumn, DistillationColumnParams

names = ['n_pentane', 'n_hexane', 'n_heptane']
thermo = IdealThermo({s: get_species_data(s) for s in names})
feed = make_stream({'n_pentane': 30., 'n_hexane': 40., 'n_heptane': 30.},
                   T=360., P=101325.)

column = DistillationColumn(
    DistillationColumnParams(species_order=names, n_stages=20,
                             feed_stage=10, P=101325.),
    thermo,
)

for use_mesh in (True, False):
    d, b, info = column(feed, R=2.0, B_spec=40.0, use_mesh=use_mesh)
    print(f"use_mesh={use_mesh}")
    for s in names:
        D, B = float(d['F_' + s]), float(b['F_' + s])
        print(f"   {s:10s} feed={float(get_flows(feed)[s]):7.3f}  D+B={D + B:7.3f}")
```

```
use_mesh=True
   n_pentane  feed= 30.000  D+B= 29.954
   n_hexane   feed= 40.000  D+B= 40.046
   n_heptane  feed= 30.000  D+B= 30.000
use_mesh=False
   n_pentane  feed= 30.000  D+B= 51.696      <- +72 %
   n_hexane   feed= 40.000  D+B=  7.601      <- -81 %
   n_heptane  feed= 30.000  D+B= 40.703      <- +36 %
```

51.7 mol/s of n-pentane leaves a column fed 30 mol/s of it.

## Where it comes from

`_solve_constant_molar_overflow` is a Lewis-Matheson sweep: it alternates an
equilibrium step with an operating-line step and normalises the composition at
each stage, but nothing in the loop enforces a component balance over the
column. The products are then formed as `D * x_D` and `B * x_B` from the
converged profile ends, so whatever the sweep's compositions come to is what
gets reported, balanced or not. The MESH solver does not have this problem
because its tridiagonal solve *is* the component material balance.

A related symptom, which may share a root cause: the CMO result is completely
insensitive to `feed_stage`. Sweeping `feed_stage` over 8, 9, 10, 11, 12 on the
case above gives byte-identical products, which cannot be right for a column
whose feed location is moving four stages.

## Why it matters

`use_mesh=False` is a documented, supported option, offered as "the faster but
less accurate CMO solution". Less accurate is expected of a shortcut; not
conserving mass is a different thing, and there is nothing in the returned
`info` that would tell a caller it happened. It is also the MESH solver's warm
start, so a bad CMO solution costs MESH iterations even when the final answer
is sound.

## Suggested fix

Either enforce the component balances in the sweep (rescale each species'
profile so `D x_{D,i} + B x_{B,i} = F z_i` at the end of each iteration is the
cheap version), or — if the sweep cannot be made to conserve mass without
becoming the MESH solve — make `use_mesh=False` report the closure error in
`info` so a caller can see it, and say plainly in the docs that the CMO path
does not close component balances.

Whichever way it goes, a test asserting `D_i + B_i == F_i` per species on both
paths would keep it honest. `tests/test_distillation.py` currently asserts that
only for the MESH path, at `rel=0.05`.

## Context

Found while auditing the distillation docs (#210 follow-up), by checking
whether `feed_stage` did anything on the CMO path.
